README.md

```diff
@@ -60,3 +62,44 @@ Then use the `analyticsClient` variable:
 ```js
 const urlParamsHandler = new analyticsClient.AnalyticsUrlParams();
 ```
+
+## Configuration
+
+We use environment configuration on the [analytics-backend](https://github.com/balena-io/analytics-backend) to mask the actual API tokens of the final destination services of these events.
+We expose easy to remember names to map the destination specific tokens to `projectName` variable used as part of client initialization. analytics-backend uses the value of `projectName` to reason where to pipe the event flow.
+The masking helps with preventing abuse and the mapping makes the configuration more flexible to support multiple final destinations.
+Reach out to the data team when in doubt which project to use or if you need to define a new project.
+
+In the final destinations, we separate staging and production environments. We use the same `projectName` value but different tokens to differentiate the production and staging environments. So you only need to update `endpoint` to pick the right destination.
+
+Use the staging environment for testing purposes by setting `endpoint` in the `createClient` initialization to `data.balena-staging.com`. Once you are happy with your tests and ready to go live, update the `endpoint` to `data.balena-cloud.com` for production environment.
+
+Additionally, identify your component using the parameter `componentName`. It's best to keep the value short yet descriptive.
+
+Please ask Operation folks for `#access` to the relevant project in your final destination so you can validate that the tracked events show as expected.
+
+Finally, there is the optional configuration parameter `componentVersion` to track the version of a component sending each event. This is useful to track changes across new component versions so we strongly recommend tracking it.
+
+## Web tracker
+
+[`WebTracker`](src/web.ts) exposes an events tracking interface suitable for web apps. It provides a simple `Page View` event tracking with page loading metrics via `trackPageView`.
```

Comment:
Again, an example on how to use it with a short explanation would be nice here. In general, `examples >>> text` 😄

Reply:
Thanks for the thoughts @sradevski. You raise a very good point. I'm adding couple of more examples to the top of the readme to illustrate the common use cases. Then is there any harm to keep such verbose explanations towards the bottom of the docs?

So the option is still there for anyone who wants to understand the internals.

Reply:
I personally don't think it is needed, at least the internal implementation parts, since every single character or word we write is something that needs to be maintained and can get outdated. The less we write, the better (of course, without losing information)

Reply:
Good point on the maintenance part Stevche. 

I prefer to keep these explanations here until we find a better place. In a sense, these are the answers to FAQ raised in flowdock threads. If similar questions keep popping up, I'll take it as a strong signal that indeed no one reads these explanations here : )  
Also, going forward, it might make sense to keep such explanations in the repo but move them to a new section for the maintainers.